### PR TITLE
add additional IAP price tiers

### PR DESCRIPTION
**Pull request by phillipthelen**
Adds new gem purchase price tiers for the apps and removes an unused one for the iOS app.

### CODE REVIEW

File: website/server/controllers/top-level/payments/iap.js [hunk 1]
```diff
@@ -54,12 +54,26 @@ api.iapAndroidVerify = {
       userId: user._id,
     });
 
-    await payments.buyGems({
-      user,
-      paymentMethod: 'IAP GooglePlay',
-      amount: 5.25,
-      headers: req.headers,
-    });
+    switch (purchaseData.productId) {
```
Comment (khipkin) [inline website/server/controllers/top-level/payments/iap.js]: Suggestion: Assign a value to `amount` in the switch, but make the API call afterwards to reduce redundancy.

Comment (paglias) [reply website/server/controllers/top-level/payments/iap.js]: agreed

Comment (crookedneighbor) [reply website/server/controllers/top-level/payments/iap.js]: Agreed.

**Comment by phillipthelen** [comment]
changed the switch statement

File: website/server/controllers/top-level/payments/iap.js [hunk 2]
```diff
@@ -54,12 +54,27 @@ api.iapAndroidVerify = {
...
+        amount = 10.5;
+        break;
+      case 'com.habitrpg.android.habitica.iap.84gems':
+        amount = 21;
+        break;
+    }
+
+    if (amount === 0) throw new Error('INVALID_ITEM_PURCHASED');
+
+    await payments.buyGems({user, paymentMethod: 'IAP GooglePlay', amount, headers: req.headers}); // eslint-disable-line babel/no-await-in-loop
```
Comment (paglias) [inline website/server/controllers/top-level/payments/iap.js]: the eslint comment should not be necessary, also can you split the object on multiple lines as it was before?

File: website/server/controllers/top-level/payments/iap.js [hunk 3]
```diff
@@ -54,12 +54,27 @@ api.iapAndroidVerify = {
       userId: user._id,
     });
 
-    await payments.buyGems({
-      user,
-      paymentMethod: 'IAP GooglePlay',
-      amount: 5.25,
-      headers: req.headers,
-    });
+    let amount = 0;
```
Comment (paglias) [inline website/server/controllers/top-level/payments/iap.js]: `let amount;` without = 0

File: website/server/controllers/top-level/payments/iap.js [hunk 4]
```diff
@@ -54,12 +54,27 @@ api.iapAndroidVerify = {
...
+        break;
+      case 'com.habitrpg.android.habitica.iap.42gems':
+        amount = 10.5;
+        break;
+      case 'com.habitrpg.android.habitica.iap.84gems':
+        amount = 21;
+        break;
+    }
+
+    if (amount === 0) throw new Error('INVALID_ITEM_PURCHASED');
```
Comment (paglias) [inline website/server/controllers/top-level/payments/iap.js]: `if (!amount)`

File: website/server/controllers/top-level/payments/iap.js [hunk 5]
```diff
@@ -114,23 +129,27 @@ api.iapiOSVerify = {
...
-            await payments.buyGems({user, paymentMethod: 'IAP AppleStore', amount: 10.5, headers: req.headers}); // eslint-disable-line babel/no-await-in-loop
+            amount = 10.5;
+            break;
+          case 'com.habitrpg.ios.Habitica.84gems':
+            amount = 21;
             break;
-          default:
-            correctReceipt = false;
         }
+        if (amount === 0) {
```
Comment (paglias) [inline website/server/controllers/top-level/payments/iap.js]: if (!amount)

File: website/server/controllers/top-level/payments/iap.js [hunk 6]
```diff
@@ -114,23 +129,27 @@ api.iapiOSVerify = {
...
+          correctReceipt = false;
+          break;
+        }
+        await payments.buyGems({user, paymentMethod: 'IAP AppleStore', amount, headers: req.headers}); // eslint-disable-line babel/no-await-in-loop
```
Comment (paglias) [inline website/server/controllers/top-level/payments/iap.js]: split in multiple lines

File: website/server/controllers/top-level/payments/iap.js [hunk 7]
```diff
@@ -114,23 +129,27 @@ api.iapiOSVerify = {
           throw new Error('RECEIPT_ALREADY_USED');
         }
 
+        let amount = 0;
```
Comment (paglias) [inline website/server/controllers/top-level/payments/iap.js]: let amount;

**Review by paglias — CHANGES_REQUESTED** [review]

**Review by paglias — APPROVED** [review]

**Pull request merged**